.travis.yml

```diff
@@ -1,9 +1,34 @@
+addons:
+  firefox: "latest"
+
 language: ruby
+sudo: false
+
+cache:
+  directories:
+    - "travis_phantomjs"
+    - "travis_geckodriver"
+
 rvm:
   - '2.3.1'
+
 before_script:
+  - "phantomjs --version"
+  - "export PATH=$PWD/travis_phantomjs/phantomjs-2.1.1-linux-x86_64/bin:$PATH"
+  - "phantomjs --version"
+  - "if [ $(phantomjs --version) != '2.1.1' ]; then rm -rf $PWD/travis_phantomjs; mkdir -p $PWD/travis_phantomjs; fi"
+  - "if [ $(phantomjs --version) != '2.1.1' ]; then wget https://assets.membergetmember.co/software/phantomjs-2.1.1-linux-x86_64.tar.bz2 -O $PWD/travis_phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2; fi"
```

Comment:
`phantomjs` will always be a valid command because the older version is installed on travis by default. 

This might want to be rethought since it is pretty weird I guess.